lib/mpool/lib/mdc_file.c

```diff
@@ -396,25 +394,26 @@ mdc_file_open(
 
     err = mdc_file_mmap(mfp, mfp->size, rdonly);
     if (err)
-        goto err_exit1;
+        goto err_exit;
 
+    /* On detecting an incomplete log header from a crash during mdc file erase,
+     * return a valid handle.
+     */
     err = mdc_file_validate(mfp, gclose, gen);
-    if (err) {
+    if (err && merr_errno(err) != ENOMSG) {
         mdc_file_unmap(mfp);
-        goto err_exit1;
+        goto err_exit;
```

Comment:
Maybe comment should say something about returning ENOMSG as well as a valid handle...